src/diffusers/loaders/lora.py

```diff
@@ -977,15 +978,17 @@ def set_adapters_for_text_encoder(
             raise ValueError("PEFT backend is required for this method.")
 
         def process_weights(adapter_names, weights):
-            if weights is None:
-                weights = [1.0] * len(adapter_names)
-            elif isinstance(weights, float):
-                weights = [weights]
+            if not isinstance(weights, list):
+                weights = [weights] * len(adapter_names)
```

Comment:
We should not be deviating from the default values here. These changes remove the default value of 1.0 from `weights`. 

Edit: I see you're tackling it a little later. Okay for me.